setup.py

```diff
@@ -112,7 +112,7 @@ def build_extensions(self):
 pyspark_require_list = ['pyspark>=2.3.2;python_version<"3.8"',
                         'pyspark>=3.0.0;python_version>="3.8"']
 # Pin h5py: https://github.com/h5py/h5py/issues/1732
-spark_require_list = ['h5py<3', 'numpy', 'petastorm>=0.9.8,<0.11.0', 'pyarrow>=0.15.0']
+spark_require_list = ['h5py<3', 'numpy', 'petastorm>=0.11.0', 'pyarrow>=0.15.0']
```

Comment:
@tgaddair do you think requiring latest petastorm for Horovod on Spark is too much of a constraint? Should we support petastorm <0.11 for a transition period?

If we go forward with this, that "limitation" must be recorded in the CHANGELOG and release notes.

Reply:
The only difference of `petastorm v0.11.0` is refactoring `pytorch inmem dataloader`, which I believe is not used frequently. Yep, I agree to add `CHANGELOG` and release notes.

Reply:
updated CHANGLOG to bring `pytorch inmem dataloder` change.

Reply:
I think this is fine. Petastorm is pretty lightweight with few overlapping dependencies, so shouldn't be an issue. Unfortunately, it's frequently the case that Horovod needs to be updated to the latest Petastorm to take advantage of new features.